src/jit/lowerxarch.cpp

```diff
@@ -1110,6 +1116,23 @@ void Lowering::TreeNodeInfoInitCall(GenTreeCall* call)
         // computed into a register.
         if (!call->IsFastTailCall())
         {
+#ifdef _TARGET_X86_
+            // On x86, we need to generate a very specific pattern for indirect VSD calls:
+            //
+            //    3-byte nop
+            //    call dword ptr [eax]
+            //
+            // Where EAX is also used as an argument to the stub dispatch helper. Make
+            // sure that the call target address is computed into EAX in this case.
+            if (((call->gtFlags & GTF_CALL_VIRT_KIND_MASK) == GTF_CALL_VIRT_STUB) && (call->gtCallType == CT_INDIRECT))
+            {
+                assert(ctrlExpr->isIndir());
+
+                ctrlExpr->gtGetOp1()->gtLsraInfo.setDstCandidates(l, RBM_EAX);
```

Comment:
>               ctrlExpr->gtGetOp1()->gtLsraInfo.setDstCandidates(l, RBM_EAX); [](start = 1, length = 77)

Here we should be setting srcCandidates, since call node is going to consume ctrlExpr operand.
The register(s) specified in srcCandidates will be used for reg allocation at the point of Use.
Due to register pressure, if the operand register is not in the right single reg set by Lower, then codegen is supposed to move it before using it.